### PR TITLE
🔨  Deprecate countries_regions.csv

### PR DESCRIPTION
**Pull request by Marigold**
Deprecate countries_regions.csv and replace it by regions dataset. Fixes https://github.com/owid/etl/issues/1081.

There's a helper function `load_regions` that loads required columns from regions dataset (can create it by joining multiple tables).

I checked all datasets with datadiff and if there are differences, they're tiny (likely because of Northern Cyprus and Namibia)

## Open questions
1. Northern Cyprus used to be in Asia continent, but now it doesn't belong to any continent which causes problems to `garden/health/2023-04-25/wgm_2018`. Should we add it to Asia?
2. We still have `reference` dataset, but it's not very useful without countries regions. I think we can just get rid of it.
3. All datasets depend on 2023-01-01 version of regions. It could be latest, but I think it's fine

### CODE REVIEW

**Comment by pabloarosado** [comment]
Hi @Marigold thanks for doing this!! I'm not sure if having a `load_regions()` function is a good idea. In the end, this is one of the issues we wanted to avoid with the old setup (loading datasets under the hood). One solution would be that `load_regions` ingests the regions dataset as a mandatory argument. But I think a much better option would be that `regions` dataset has a table with all necessary data in it.

We discussed it when designing the `regions` dataset and decided to have separate tables, each with their own data. But in practice we are almost always going to have to join different tables (e.g. codes and region names). So maybe the best solution is to add a `regions` table to the `regions` dataset, containing everything we need (instead of or in addition to `tb_definitions, tb_aliases, tb_members, tb_related, tb_transitions, tb_legacy_codes`). What do you think? Maybe we can briefly discuss it in the data architecture chat.

**Comment by pabloarosado** [comment]
In a similar vein, I think the function `country_code_to_country` (which seems to only be used by two data steps) should not exist.

**Comment by pabloarosado** [comment]
Regarding your questions:

> 1. Northern Cyprus used to be in Asia continent, but now it doesn't belong to any continent which causes problems to garden/health/2023-04-25/wgm_2018. Should we add it to Asia?

Yes, I suppose that makes sense.

> 2. We still have reference dataset, but it's not very useful without countries regions. I think we can just get rid of it.

Yes, it's time to kill it.

> 3. All datasets depend on 2023-01-01 version of regions. It could be latest, but I think it's fine

I think having a version is the better approach.

**Comment by Marigold** [comment]
@pabloarosado implemented all your suggestions, datadiff looks good, only minor differences there (here's [an old one](https://buildkite.com/our-world-in-data/etl-build-public-datasets-branch/builds/2414#018842c5-7fe4-4442-b1a6-59e14cf8179d) if the current one hasn't finished yet. It shows an issue with `lis` datasets, but that have been fixed by the latest commit). Could you review / approve please?

File: etl/paths.py [hunk 1]
```diff
@@ -8,7 +8,11 @@
 DATA_DIR = BASE_DIR / "data"
 SNAPSHOTS_DIR = BASE_DIR / "snapshots"
 STEP_DIR = BASE_DIR / "etl" / "steps"
-REFERENCE_DATASET = DATA_DIR / "garden" / "reference"
+
+# Regions paths
+REGIONS_DATASET_PATH = DATA_DIR / "garden" / "regions" / "2023-01-01" / "regions"
+LATEST_REGIONS_REGIONS_YML = sorted((STEP_DIR / "data/garden/regions/").glob("*/regions.yml"))[-1]
```
[the remark this answers is not in the document]
Comment (Marigold) [reply etl/paths.py:14]: Right! Removed `REGIONS_DATASET_PATH` and using `LATEST_REGIONS_REGIONS_YML` everywhere.

File: etl/paths.py [hunk 2]
```diff
@@ -8,7 +8,11 @@
...
+LATEST_REGIONS_DATASET_PATH = BASE_DIR / LATEST_REGIONS_REGIONS_YML.relative_to(STEP_DIR).with_suffix("")
```
[the remark this answers is not in the document]
Comment (Marigold) [reply etl/paths.py:16]: I'll add it "just in time" when we need it (couldn't find good use case for it).

**Pull request merged**